### PR TITLE
Save raw content in addition to readability output

### PR DESCRIPTION
**Pull request by jasonrdsouza**
To allow for archival of PDF and other non-text formats, as well as a
backup in case the readability library mangles an article.

### CODE REVIEW

File: milton-processor/scribe.ts [hunk 1]
```diff
@@ -46,13 +47,18 @@ app.post("/simplify", async (req, res) => {
...
-    readability.fetchArticle(url).then((response) => {
+    fetcher.fetchArticle(url).then((response) => {
       console.log(`Saving contents of ${url}`);
-        const po: readability.ParsedOutput = readability.parse(response, url);
+
+      if (response.isTextual()) {
+        // save raw contents as text and perform readability parsing
+        cache.saveRawContents(url, response.body, response.contentType);
+
+        const po: readability.ParsedOutput = readability.parse(response.body, url);
```
Comment (MattX) [inline milton-processor/scribe.ts:61]: It might make sense to only process `text/html` pages through readability

File: milton-processor/cache.ts [hunk 2]
```diff
@@ -49,42 +52,67 @@ function sha256Hash(url: string): string {
...
+  }).catch((err) => {
+    console.error(`Failed to save raw contents of URL (${url}) with error: ${err.message}`);
+  });
+}
+
+export function fetchRawContents(url: string): Promise<string> {
+  const file = urlToFile(url, RAW_CONTENTS_PATH);
+  const cachedContents = file.download().then((data) => {
+    const fileContents = data[0].toString();
+    return fileContents;
```
Comment (MattX) [inline milton-processor/cache.ts:84]: You can elide the variable here (and make this lambda into a single expression)

File: milton-processor/scribe.ts (hunk 1 above)
Comment (jasonrdsouza) [reply milton-processor/scribe.ts:61]: fixed

File: milton-processor/cache.ts (hunk 2 above)
Comment (jasonrdsouza) [reply milton-processor/cache.ts]: updated

**Pull request merged**